polaris-pipeline/coordinator/Providers/OrchestrationProvider.cs

```diff
@@ -0,0 +1,335 @@
+            return new HttpResponseMessage(HttpStatusCode.Locked);
+        }
+
+        await orchestrationClient.StartNewAsync(nameof(RefreshCaseOrchestrator), instanceId, casePayload);
+
+        _logger.LogMethodFlow(correlationId, loggingName, $"{nameof(OrchestrationProvider)} Succeeded - Started {nameof(RefreshCaseOrchestrator)} with instance id '{instanceId}'");
+        return orchestrationClient.CreateCheckStatusResponse(req, instanceId);
+    }
+
+    public async Task<HttpResponseMessage> DeleteCaseAsync(IDurableOrchestrationClient orchestrationClient, Guid correlationId, 
+        int caseId)
+    {
+        var telemetryEvent = new DeletedCaseEvent(
+                correlationId: correlationId,
+                caseId: caseId,
+                startTime: DateTime.UtcNow
+        );
+        var caseIdAsString = caseId.ToString();
+        
+        try
+        {
+            await _searchIndexService.RemoveCaseIndexEntriesAsync(caseId, correlationId);
+            telemetryEvent.RemovedCaseIndexTime = DateTime.UtcNow;
+
+            await _searchIndexService.WaitForCaseEmptyResultsAsync(caseId, correlationId);
+            telemetryEvent.IndexSettledTime = DateTime.UtcNow;
+            
+            await _blobStorageService.DeleteBlobsByCaseAsync(caseIdAsString, correlationId);
+            telemetryEvent.BlobsDeletedTime = DateTime.UtcNow;
+
+            // Terminate Orchestrations (can't terminate Durable Entities with Netherite backend, but can Purge - see below)
+            var terminateOrchestrationQueries = GetOrchestrationQueries(_terminateStatuses, caseIdAsString);
+            var terminateOrchestrationInstanceIds = await TerminateOrchestrations(orchestrationClient, terminateOrchestrationQueries, correlationId);
+            telemetryEvent.TerminatedInstancesCount = terminateOrchestrationInstanceIds.Count;
+            telemetryEvent.GotTerminateInstancesTime = DateTime.UtcNow;
+            telemetryEvent.TerminatedInstancesTime = DateTime.UtcNow;
+
+            // Purge Orchestrations and Durable Entities
+            var purgeConditions = GetOrchestrationQueries(_purgeStatuses, caseIdAsString);
+            purgeConditions.AddRange(GetDurableEntityQueries(_terminateStatuses, caseIdAsString));
+            var success = await Purge(orchestrationClient, purgeConditions, correlationId);
+            telemetryEvent.EndTime = DateTime.UtcNow;
+
+            _telemetryClient.TrackEvent(telemetryEvent);
+
+            return new HttpResponseMessage(success ? HttpStatusCode.OK : HttpStatusCode.InternalServerError);
+        }
+        catch (Exception)
+        {
+            _telemetryClient.TrackEventFailure(telemetryEvent);
+            throw;
+        }
+    }
+
+    public async Task<HttpResponseMessage> UpdateTrackerAsync(IDurableOrchestrationClient orchestrationClient, Guid correlationId, string caseId, CaseOrchestrationPayload casePayload, HttpRequestMessage req)
```

Comment:
Might be good to remove this method now. It was used in testing to support the manual setting of the tracker from Postman, but that isn't possible now it's created from multiple source entities

Reply:
Done